tests/loganalyzer/loganalyzer.py

```diff
@@ -0,0 +1,193 @@
+COMMON_MATCH = join(split(__file__)[0], "loganalyzer_common_match.txt")
+COMMON_IGNORE = join(split(__file__)[0], "loganalyzer_common_ignore.txt")
+COMMON_EXPECT = join(split(__file__)[0], "loganalyzer_common_expect.txt")
+SYSLOG_TMP_FOLDER = "/tmp/pytest-run/syslog"
+
+
+class LogAnalyzer:
+    def __init__(self, ansible_host, marker_prefix, dut_run_dir="/tmp"):
+        self.ansible_host = ansible_host
+        self.dut_run_dir = dut_run_dir
+        self.extracted_syslog = os.path.join(self.dut_run_dir, "syslog")
+        self.marker_prefix = marker_prefix
+        self.ansible_loganalyzer = ansible_loganalyzer(self.marker_prefix, False)
+
+        self.match_regex = []
+        self.expect_regex = []
+        self.ignore_regex = []
+
+    def _add_end_marker(self, marker):
+        """
+        @summary: Add stop marker into syslog on the DUT.
+
+        @return: True for successfull execution False otherwise
+        """
+        self.ansible_host.copy(src=ANSIBLE_LOGANALYZER_MODULE, dest=os.path.join(self.dut_run_dir, "loganalyzer.py"))
+
+        cmd = "python {run_dir}/loganalyzer.py --action add_end_marker --run_id {marker}".format(run_dir=self.dut_run_dir, marker=marker)
+
+        logging.debug("Adding end marker '{}'".format(marker))
+        self.ansible_host.command(cmd)
+
+    def update_marker_prefix(self, marker_prefix):
+        """
+        @summary: Update configured marker prefix
+        """
+        self.marker_prefix = marker_prefix
+
+    def load_common_config(self):
+        """
+        @summary: Load regular expressions from common files, which are localted in folder with legacy loganalyzer.
+                  Loaded regular expressions are used by "analyze" method to match expected text in the downloaded log file.
+        """
+        self.match_regex = self.ansible_loganalyzer.create_msg_regex([COMMON_MATCH])[1]
+        self.ignore_regex = self.ansible_loganalyzer.create_msg_regex([COMMON_IGNORE])[1]
+        self.expect_regex = self.ansible_loganalyzer.create_msg_regex([COMMON_EXPECT])[1]
+
+    def parse_regexp_file(self, src):
+        """
+        @summary: Get regular expressions defined in src file.
+        """
+        return self.ansible_loganalyzer.create_msg_regex([src])[1]
+
+    def run_cmd(self, callback, *args, **kwargs):
+        """
+        @summary: Initialize loganalyzer, execute function and analyze syslog.
+
+        @param callback: Python callable or function to be executed.
+        @param args: Input arguments for callback function.
+        @param kwargs: Input arguments for callback function.
+
+        @return: Tuple of two items: (callback(*args, **kwargs) -> VALUE, self.analyze() -> dict)
+        """
+        marker = self.init()
+        try:
+            call_result = callback(*args, **kwargs)
+        except Exception as err:
+            logging.error("Error during callback execution:\n{}".format(err))
+            logging.debug("Log alaysis result\n".format(self.analyze()))

```

Comment:
Below analyze requires marker and here it does not? Is this a typo?

Reply:
Fixed.